content/docs/concepts/architecture.md

```diff
@@ -0,0 +1,328 @@
+
+![Composition](/img/concepts/architecture/composition.svg)
+
+```yaml
+---
+apiVersion: networkservicemesh.io/v1
+kind: NetworkService
+metadata:
+  name: service-mesh@finance.example.com
+spec:
+  payload: IP
+  matches:
+    - source_selector:
+        service: envoy-proxy
+      routes:
+        - destination_selector:
+            service: vl3
+    - source_selector:
+      routes:
+      - destination_selector:
+          service: envoy-proxy
+```
+
+Please note: there is nothing magic about the choice of labels as 'service: ...' as with all labels, the choice is arbitrary,
+its the matching that matters.
+
+#### Selective Composition
+
+Sometimes it is desirable to have different {{<color "#0050EF" >}}Clients{{< /color >}} receive a different composition of {{<color "#008A00" >}}Endpoints{{< /color >}} to provide a Network Service.
+
+For example, imagine that a {{<color "#0050EF" >}}Client{{< /color >}} is version v1.1 of an app foo.  It is known that v1.1 of app foo has a security
+vulnerability.  There is a plan to remediate to foo version v1.2 with the fix.  The schedule for that is six weeks out.
+App foo needs to stay in deployment in the interim.
+
+An expensive IPS can provide protection from the vulnerability.  By keying off of labels provided by the {{<color "#0050EF" >}}Clients{{< /color >}} when
+they Request the Network Service, NSM can selectively interpose an IPS between all instances of foo v1.1 and the vL3
+for the Network Service.
+
+![Selective Composition](/img/concepts/architecture/selective-composition.svg)
+
+```yaml
+---
+apiVersion: networkservicemesh.io/v1
+kind: NetworkService
+metadata:
+  name: secure-vl3@sales.example.com
+spec:
+  payload: IP
+  matches:
+    - source_selector:
+        app: foo
+        version: "v1.1"
+      routes:
+        - destination_selector:
+            provides: ips
+    - source_selector:
+        provided: ips
+      routes:
+        - destination_selector:
+            provides: vl3
+    - source_selector:
+        routes:
+        - destination_selector:
+            provides: vl3
+```
+
+All other workloads using the Network Service continue normally without the IPS.
+
+Please note: there is nothing magic about the choice of labels as:
+
+- provides
+- provided
+- app
+- version
+
+as with all labels, the choice is arbitrary, it's the matching that matters.
+
+### Topologically Aware Endpoint Selection
+
+Topology for both {{<color "#0050EF" >}}Clients{{< /color >}} and {{<color "#008A00" >}}Endpoints{{< /color >}} can be expressed by source or destination labels.  Examples:
+
+- nodeName
+- clusterName
+- zone
+- cloudProvider
+
+etc.
+
+Network Service Mesh supports dynamic specification of destination labels based on the source labels in the Request.
+In a destination_selector {{ .${labelName} }} will substitute in the value for $labelName from the source labels of the Request.
```

Comment:
>{{ .${labelName} }}
>$labelName

I guess this is an analogue to shell var substitution. This was not obvious to me from the first glance, it looked like a part of NSM syntax.
I think it would be easier to read if we dropped the `${}` part.

Reply:
Fixed.  Let me know how the fix reads to you.

Reply:
The fix looks great!